service/gcs/bridge/bridge.go

```diff
@@ -268,15 +268,15 @@ func (b *bridge) shutdownContainer(message []byte) (*prot.MessageResponseBase, e
 	return response, nil
 }
 
-func (b *bridge) terminateProcess(message []byte) (*prot.MessageResponseBase, error) {
+func (b *bridge) signalProcess(message []byte) (*prot.MessageResponseBase, error) {
 	response := newResponseBase()
-	var request prot.ContainerTerminateProcess
+	var request prot.ContainerSignalProcess
 	if err := commonutils.UnmarshalJSONWithHresult(message, &request); err != nil {
 		return response, errors.Wrapf(err, "failed to unmarshal JSON for message \"%s\"", message)
 	}
 	response.ActivityID = request.ActivityID
 
-	if err := b.coreint.TerminateProcess(int(request.ProcessID)); err != nil {
+	if err := b.coreint.SignalProcess(int(request.ProcessID), request.Options); err != nil {
```

Comment:
That's fair. My reasoning for not doing this is Go doesn't entirely support optional JSON fields, and the best way (at least the best way I know, there might be better ways) of handling this would be to use json.RawMessage as the field type for Signal in order to detect if it's present or not. Either that or special case the default 0 Signal value to mean kill. Both of these seemed hacky enough that it wasn't worth putting them into the GCS to support older RS3 Windows versions. However, if we don't want a breaking change, I can put those changes into the GCS, maybe temporarily even.